src/declarativewebcontainer.h

```diff
@@ -182,12 +188,18 @@ private slots:
     void updateTitle(const QString &newTitle);
     bool canInitialize() const;
     void loadTab(int tabId, QString url, QString title, bool force);
+    void updateMode();
 
     QPointer<DeclarativeWebPage> m_webPage;
     QPointer<DeclarativeTabModel> m_model;
     QPointer<QQmlComponent> m_webPageComponent;
     QPointer<SettingManager> m_settingManager;
-    QScopedPointer<WebPages> m_webPages;
+    QPointer<WebPages> m_webPages;
+
+    QScopedPointer<DeclarativeTabModel> m_persistentTabModel;
+    QScopedPointer<DeclarativeTabModel> m_privateTabModel;
```

Comment:
Setting parent for scoped members was a bad idea.
Please replace QScopedPointer with QPointer. Scoping is not needed if parent is set for an instance. Otherwise Qt will try to delete the member object twice when DeclarativeWebContainer instance is destroying => crash.